src/syscall/fs_wasip1.go

```diff
@@ -750,44 +750,52 @@ func Fsync(fd int) error {
 	return errnoErr(errno)
 }
 
-func makeIOVec(b []byte) *iovec {
-	return &iovec{
+func bytesPointer(b []byte) unsafe.Pointer {
+	return unsafe.Pointer(unsafe.SliceData(b))
+}
+
+func stringPointer(s string) unsafe.Pointer {
+	return unsafe.Pointer(unsafe.StringData(s))
+}
+
+func makeIOVec(b []byte) unsafe.Pointer {
+	return unsafe.Pointer(&iovec{
 		buf:    uintptr32(uintptr(unsafe.Pointer(unsafe.SliceData(b)))),
```

Comment:
Could this be
```suggestion
		buf:    uintptr32(uintptr(bytesPointer(b)),
```
now?

Reply:
Yes, I wanted to verify that it still remains within the inline cost threshold so it doesn't end up causing heap allocations. I may add a test to validate it.

Reply:
Yep, definitely should make a test for it 👍🏻. Also add a comment.